### PR TITLE
(Spinbox ) Add function to set cursor to specific position

### PR DESCRIPTION
**Pull request by dyktronix**
### Description of the feature or fix

The pull request brings in a new function, that allows you to jump the cursors in a spinbox to an specific position for edition.

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update [CHANGELOG.md](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md)
- [x] Update the documentation

### CODE REVIEW

File: src/extra/widgets/spinbox/lv_spinbox.c [hunk 1]
```diff
@@ -145,6 +145,23 @@ void lv_spinbox_set_range(lv_obj_t * obj, int32_t range_min, int32_t range_max)
...
+ * @param spinbox pointer to spinbox
+ * @param pos selected position in spinbox
+ */
+void lv_spinbox_set_pos(lv_obj_t * obj, uint8_t pos)
+{
+
+    lv_spinbox_t * spinbox = (lv_spinbox_t *)obj;
+    int32_t step_limit;
+    step_limit       = LV_MAX(spinbox->range_max, (spinbox->range_min < 0 ? (-spinbox->range_min) : spinbox->range_min));
+    int32_t new_step = spinbox->step * exp10(pos);
```
Comment (embeddedt) [inline src/extra/widgets/spinbox/lv_spinbox.c]: `exp10` appears to be a floating-point function, which for most purposes is not permitted within LVGL.

File: src/extra/widgets/spinbox/lv_spinbox.c [hunk 2]
```diff
@@ -145,6 +145,23 @@ void lv_spinbox_set_range(lv_obj_t * obj, int32_t range_min, int32_t range_max)
...
+    if(pos <= 0) spinbox->step = 1;
+	else if(new_step <= step_limit) spinbox->step = new_step;
```
Comment (embeddedt) [inline src/extra/widgets/spinbox/lv_spinbox.c]: Please fix the indentation here.

**Review by embeddedt — CHANGES_REQUESTED** [review]
Thanks - just two comments.

File: src/extra/widgets/spinbox/lv_spinbox.c (hunk 2 above)
Comment (dyktronix) [reply src/extra/widgets/spinbox/lv_spinbox.c]: Hm, I'm not quiet sure why, in my IDE an in my github fork is the indentation correct. Only here it's wrong.

Comment (embeddedt) [reply src/extra/widgets/spinbox/lv_spinbox.c]: I think you may have a tab on the `else` line, but the rest of the file uses four spaces.

(And yes, GitHub is strange: the indentation even differs between this main page and the "Files changed" page. On one, the `else` is indented past the `if`, and on the other, it has less indentation than the `if`.)

**Review by embeddedt — APPROVED** [review]

**Comment by kisvegabor** [comment]
Thanks! Please add a sentence about it in the docs:
https://github.com/lvgl/lvgl/blob/master/docs/widgets/extra/spinbox.md

**Comment by kisvegabor** [comment]
Merged, thank you!

**Pull request merged**